nvm.sh

```diff
@@ -2317,6 +2381,53 @@ nvm_npmrc_bad_news_bears() {
   return 1
 }
 
+# args: path to check and transform if needed
+nvm_ensure_posix_path() {
+  [ -z "${1-}" ] && nvm_err "Need the location as first parameter" && return 1
+  local location
+  location="$1"
+
+  # If already posix path, just return it
+  [ "$(command printf '%s' "$location" | command cut -b 1)" = "/" ] && nvm_echo "$location" && return 0
+
+  # If absolute path (eg. starting with C:)
+  if command printf '%s' "$location" | command grep '^[a-zA-Z]:' >/dev/null 2>&1; then
+    # Get lower case drive letter, eg. c for C:\...
+    local letter
+    letter=$(command printf '%s' "$location" | command cut -b 1 | command tr '[:upper:]' '[:lower:]')
+    # Get upper case drive letter, eg. C for C:\...
+    local letter_upper
+    letter_upper=$(command printf '%s' "$letter" | command tr '[:lower:]' '[:upper:]')
+    local letter_prefix
+    if [ -d '/mnt/' ]; then
+      letter_prefix='/mnt'
+    elif [ -d '/drives/' ]; then
+      letter_prefix='/drives'
+    fi
+
+    # prefix like /mnt/c, /drives/c or /c
+    if [ -d "$letter_prefix/$letter" ]; then
+      letter_prefix="$letter_prefix/$letter"
+    # prefix like /mnt/C, /drives/C or /C
+    elif [ -d "$letter_prefix/$letter_upper" ]; then
+      letter_prefix="$letter_prefix/$letter_upper"
+    else
+      # Not possible to unit test properly this case
+      # nvm_err "Unable to convert the path to posix"
+      # return 1
```

Comment:
As the unit tests are running on linux and it seems not possible to create folders like /mnt/c or /c to simulate it, it is not possible to check the existence of such folder.
Should I change travis config to be able to run this test under Windows ?
Or maybe running this test under Windows GitHub action ?

Reply:
it should be possible to create folders like this - even if we’d have to mount a random folder as “c”. What happens if we try?

Reply:
It says access denied, it does not seem that we have the rights to create such folder. I ll check other tests using sudo and check again though.